test/gui/shared/steps/steps.py

```diff
@@ -920,9 +919,28 @@ def step(context, resource, group):
     sharingDialog.selectCollaborator(group, True)
 
 
+def overwriteFile(resource, content):
+    f = open(resource, "w")
+    f.write(content)
+    f.close()
+
+
+def tryToOverwriteFile(context, resource, content):
+    waitForFileToBeSynced(context, resource)
+    waitForFolderToBeSynced(context, '/')
+
+    try:
+        overwriteFile(resource, content)
+    except:
+        pass
```

Comment:
why would that ever fail? Does the client set read-only permissions on the local file?

Reply:
I found out that when a folder is shared with `read` permission, files inside it have `-r--r--r--` permissions but the folder itself has `drwxrwxr-x` permissions in the files system.
And when a file is shared with `read` permission, the file has `-rw-rw-r--` permissions

Reply:
same happens when you do it manually, or only in the test?

Reply:
Yeah, it is the same but only if we share-edit-permissions first from the server and then add the user to the client.